packages/venia-concept/src/components/ToastContainer/toastContainer.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { shape, string } from 'prop-types';
+import { useToastStore } from '@magento/peregrine/src/Toasts/context';
+import Toast from './toast';
+import { mergeClasses } from 'src/classify';
+import defaultClasses from './toastContainer.css';
+
+const ToastContainer = props => {
+    const classes = mergeClasses(defaultClasses, props.classes);
+    const { toasts } = useToastStore();
+
+    // Given a map of toasts each with a property "timestamp", sort and display
+    // based on the timestamp.
```

Comment:
Should the responsibility of sorting be the responsibility of the display or the store? The store is a more opinionated location but in this case it seems like consumers of the store would probably like to just iterate over a sorted array for display purposes rather than have to write their own sorting function.

@supernova-at @jimbo @tjwiebell what do ya'll think?

Reply:
In general, I think the store should keep records in a canonical order—in this case, the order in which the toasts are added. Since we're storing toasts in a map, that happens to be the natural order anyway. If a view wants to sort them differently, that should be the prerogative of the view.

Reply:
First of all, there should never be enough toasts that sorting is a significant cost. Definitely sort them by timestamp! They can re-sort if they want, and they should never want to.

Also: it seems like these next few lines are doing the same thing as `Array#sort(comparator)`. How about this?

```js
    const toastList = toasts.slice()
        .sort((a, b) => a.timestamp - b.timestamp)
        .map(toast => {
            // Use a random key to trigger a recreation of this component if it
            // is a duplicate so that we can re-trigger the blink animation.
            const key = toast.isDuplicate ? Math.random() : toast.id;
            return <Toast key={key} {...toast} />;
        });
```

Seems like it does the same thing unless I'm missing something. This should still go in the store, though 🙃